### PR TITLE
Threads appearance tweaks

### PR DESCRIPTION
**Pull request by germain-gg**
Following some of @janogarcia review

---
This PR currently has no changelog labels, so will not be included in changelogs.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.

Preview: https://pr8233--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/views/right_panel/_ThreadPanel.scss [hunk 1]
```diff
@@ -38,7 +42,7 @@ limitations under the License.
         }
     }
 
-    .mx_BaseCard_back ~ .mx_ThreadPanel__header {
+    .mx_BaseCard_back~.mx_ThreadPanel__header {
```
Comment (turt2live) [inline res/css/views/right_panel/_ThreadPanel.scss:45]: this change seems unrelated (and more complex to follow - will add to our style guide)

Comment (gsouquet) [reply res/css/views/right_panel/_ThreadPanel.scss:45]: Gahh, sorry, my VSCode is going crazy!

**Review by janogarcia — CHANGES_REQUESTED** [review]
@gsouquet Some regressions seem to have been introduced:

- [x] 🐛 Regression: Unread badges in the room header are not following the design spec anymore.
- [ ] 🐛 Regression: Thread panel now has uneven border widths: 8px (left), 6px (right)
